components/net/fetch/methods.rs

```diff
@@ -177,35 +176,46 @@ pub fn main_fetch(request: &mut Request,
 
         if (same_origin && !cors_flag ) ||
             current_url.scheme() == "data" ||
-            current_url.scheme() == "file" ||
```

Comment:
I am 99% sure this is going to hit the same problem as https://github.com/servo/servo/pull/14515#issuecomment-266173937 .